### PR TITLE
44 panoptic docs

### LINKED ISSUE #2: Doc: Alonet - Detr API
Complete the documentation of DETR:

* Models
* Architecture
* Training
* Data connectors

### PR DESCRIPTION
**Pull request by Johansmm**
New documentations:
* Detr_panoptic
* Panoptic callbacks
* LitPanopticDetr
* Panoptic train tutorial
* PanopticDetrR50 API

Update documentation of:  
 * Detr modules (detr, detr-r50 , detrfinetune, train, data_modules, ..)
 * Callbacks
 * Metrics
 * Criterion

And fix link by new DataConnectors modules (explained in #89 )

**Important note: Not merge until merge first #89 ** 

Closes #2 - #44 - #104

### CODE REVIEW

**Review by thibo73800 — CHANGES_REQUESTED** [review]

File: alodataset/coco_panoptic_dataset.py [hunk 1]
```diff
@@ -44,6 +45,9 @@ class CocoPanopticDataset(BaseDataset, SplitMixin):
         Include masks labels in the output, by default True
     classes : list, optional
         List of classes to be filtered in the annotation reading process, by default None
+    fix_classes_len : int, optional
+        Fix to a specific number the number of classes, filling the rest with "N/A" value.
+        Use when the number of model outputs does not match with the number of classes in the dataset, by default 250
```
[the remark this answers is not in the document]
Comment (Johansmm) [reply alodataset/coco_panoptic_dataset.py:50]: @thibo73800 Ok, I will change in #43

File: alonet/callbacks/base_metrics_callback.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+"""
+import pytorch_lightning as pl
+import aloscene
+from alonet import metrics
+from pytorch_lightning.utilities import rank_zero_only
+
+# import wandb
+
+
+class BaseMetricsCallback(pl.Callback):
```
[the remark this answers is not in the document]
Comment (Johansmm) [reply alonet/callbacks/base_metrics_callback.py:15]: @thibo73800 maybe `InstanceBasedMetricsCallbacks` ? The idea with this class is use "any" metric that have the "add_sample" behavior: Add samples in each step and used them to summary the results. But I agree with you. `InstanceBasedMetricsCallbacks` is ok?

**Comment by Johansmm** [comment]
@thibo73800: For this branch, is required merge first #89 to see the docs changes  only

File: alodataset/coco_panoptic_dataset.py [hunk 3]
```diff
@@ -101,6 +111,23 @@ def __init__(
                     items.append(self.items[i])
             self.items = items
 
+            # Fix label_types
```
[the remark this answers is not in the document]
Comment (Johansmm) [reply alodataset/coco_panoptic_dataset.py:114]: Done

File: alonet/metrics/compute_map.py [hunk 4]
```diff
@@ -1,41 +1,120 @@
-from scipy.special import softmax
-import matplotlib.pyplot as plt
-from itertools import product
+"""Stores all the information necessary to calculate the AP for one IoU and one class.
+
+Examples
```
[the remark this answers is not in the document]
Comment (Johansmm) [reply alonet/metrics/compute_map.py:3]: Done

File: alonet/metrics/compute_pq.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
+"""Compute Panoptic, Segmentation and Recognition Qualities Metrics.
```
[the remark this answers is not in the document]
Comment (Johansmm) [reply alonet/metrics/compute_pq.py:1]: Done

**Comment by thibo73800** [comment]
Thanks for all the documentation, can  you add detr_panoptic to the main readme along with a specific readme in the detr_panoptic folder ? (Following other models example)

**Review by thibo73800 — CHANGES_REQUESTED** [review]
Thanks for all the documentation, can you add detr_panoptic to the main readme along with a specific readme in the detr_panoptic folder ? (Following other models example)

**Review by thibo73800 — APPROVED** [review]

**Pull request merged**